niemet0502/Thot
# 28 web note handling

## Note handle on the web app [[Issue]](https://github.com/niemet0502/Thot/issues/28)

## Description
Enable users to create, update and delete notes.  

## Migrations 
YES

## Changes 
Describe How the solution of the feature was implemented

- Added a createdAt and updatedAt as attributes in the note database table 
- Added a project object in the note entity for each query 
- Installed packages related to the markdown codemirror, codemirror preview and so on
- Allow user to create, edit and delete note
- Created a toast component, and also a toast provider 
- Created utils to handle dates 

## Steps to Test or Reproduce
- Run the server and the web app 
- Sign up
- try to create, update and delete note
- On the note details page update the title and content then click outside, check if the markdown preview is displayed 

## Screen (Optionnal)

### Notes list 
![Capture d’écran 2023-03-17 160612](https://user-images.githubusercontent.com/62050249/225960579-5dffd341-2832-48a2-95f0-6726b14a772c.png)

### Notes Details 
![Capture d’écran 2023-03-17 160422](https://user-images.githubusercontent.com/62050249/225960607-5aea54fb-02ea-422b-ae50-4296be474c45.png)

![Capture d’écran 2023-03-17 161744](https://user-images.githubusercontent.com/62050249/225960880-93d73b96-22a9-471f-8ffc-f991ffdaf913.png)
